src/foam/nanos/http/ServiceWebAgent.java

```diff
@@ -64,11 +67,11 @@ public void execute(X x) {
       buffer_.rewind();
 
       FObject result = requestContext.create(JSONParser.class).parseString(buffer_.toString());
-
       if ( result == null ) {
         resp.setStatus(resp.SC_BAD_REQUEST);
-        System.err.println("Failed to parse request");
-        out.print("Failed to parse request: " + buffer_.toString());
+        String message = getParsingError(x, buffer_.toString());
+        System.err.println(message);
```

Comment:
Log this so that it's also saved to the log file, rather than printing to System.err. Also, do the same in the JDAO if I missed it there.

Reply:
Okay thanks. Changed it